users/views.py

```diff
@@ -57,6 +58,64 @@ def logout(self, request):
         data = {'success': 'Sucessfully logged out'}
         return Response(data=data, status=status.HTTP_200_OK)
 
+    @action(methods=['POST'], detail=False, url_path='changepassword', permission_classes=[IsAuthenticated])
+    def password_change(self, request):
+        serializer = self.get_serializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        request.user.set_password(serializer.validated_data['new_password'])
+        request.user.save()
+        return Response(status=status.HTTP_204_NO_CONTENT)
+
+    def get_serializer_class(self):
+        if not isinstance(self.serializer_classes, dict):
+            raise ImproperlyConfigured("serializer_classes should be a dict mapping.")
+
+        if self.action in self.serializer_classes.keys():
+            return self.serializer_classes[self.action]
+        return super().get_serializer_class()
+
+class ProfileViewSet(viewsets.GenericViewSet):    
+    permission_classes = [IsAuthenticated, ]
+    serializer_classes = {
+        'profile': serializers.UserProfileSerializer,
+        'profile_picture': serializers.UserProfileSerializer,
+    }
+
+    @action(methods=['GET', 'PUT'], detail=False)
+    def profile(self, request):
+        if request.method == 'GET':
+            user = User.objects.get(pk=request.user.id)
+            user_serializer = serializers.UserProfileSerializer(user).data
+            return Response(data=user_serializer, status=status.HTTP_200_OK)
+        elif request.method == 'PUT':
+            profile_serializer = self.get_serializer(data=request.data)
+            profile_serializer.is_valid(raise_exception=True)
+            try:
+                user = User.objects.filter(pk=request.user.id).update(**profile_serializer.validated_data)
+                print(user)
+            except ObjectDoesNotExist:
+                data = {'error': 'User not found'}
+                return Response(data, status=status.HTTP_404_NOT_FOUND)
+            data = {'success': 'Update profile successfully'}
+            return Response(data, status=status.HTTP_204_NO_CONTENT)
+
+    @action(methods=['POST'], detail=False, url_path='profile/picture')
+    def profile_picture(self, request):
+        try:
+            file = request.data['file']
+            user = User.objects.get(pk=request.user.id)
```

Comment:
write a function `get_by_id` in `UserService` to handle get user and raise user not found